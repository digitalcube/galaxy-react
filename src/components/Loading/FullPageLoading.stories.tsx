import React, { useEffect } from 'react';
import { Meta, Story } from '@storybook/react';
import { FullPageLoading, FullPageLoadingProps } from './FullPageLoading';
import { useState } from 'react';

const meta: Meta = {
  title: 'Components/Loading/FullPageLoading',
  component: FullPageLoading,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<FullPageLoadingProps> = (args) => {
  const [show, shouldShow] = useState(args.show);

  useEffect(() => {
    shouldShow(args.show);
  }, [args.show, shouldShow]);

  return (
    <>
      <FullPageLoading
        {...{
          show,
          shouldShow,
        }}
      />
      <button
        onClick={() => {
          shouldShow(true);
        }}
      >
        Show
      </button>
    </>
  );
};

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  show: true,
};
