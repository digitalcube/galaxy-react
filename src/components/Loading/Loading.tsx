import React, { FC } from 'react';
import { Spinner, SpinnerProps } from 'reactstrap';

export type LoadingProps = {
  show: boolean;
  loadingText?: string;
} & SpinnerProps;

export const Loading: FC<LoadingProps> = ({
  show,
  loadingText = 'Loading...',
  ...spinnerProps
}) => {
  if (!show) return null;

  const props: SpinnerProps = spinnerProps;
  if (props.sizes) props.sizes = 'sm';
  props.className = props.className || 'mr-2';
  return (
    <>
      <Spinner {...props}></Spinner>
      {loadingText}
    </>
  );
};
