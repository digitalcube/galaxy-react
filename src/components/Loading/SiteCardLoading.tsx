import React, { FC } from 'react';
import ContentLoader from 'react-content-loader';
import { SiteCard } from '../SiteCard';

export type SiteCardLoadingProps = {
  show?: boolean;
  className?: string;
};

export const SiteCardLoading: FC<SiteCardLoadingProps> = ({
  show = true,
  className,
}) => {
  if (!show) return null;

  return (
    <SiteCard className={className}>
      <div className="p-3">
        <ContentLoader height={80} speed={2}>
          <rect x="90" y="5" rx="4" ry="4" width="200" height="10" />
          <rect x="90" y="25" rx="3" ry="3" width="250" height="10" />
          <rect x="90" y="45" rx="3" ry="3" width="150" height="10" />
          <circle cx="40" cy="40" r="40" />
        </ContentLoader>
      </div>
    </SiteCard>
  );
};
