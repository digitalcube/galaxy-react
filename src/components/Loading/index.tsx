export * from './Loading';
export * from './FullPageLoading';
export * from './SiteCardLoading';
