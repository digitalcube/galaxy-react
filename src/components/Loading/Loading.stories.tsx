import React, { useEffect } from 'react';
import { Meta, Story } from '@storybook/react';
import { Loading, LoadingProps } from './Loading';
import { useState } from 'react';

const meta: Meta = {
  title: 'Components/Loading',
  component: Loading,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<LoadingProps> = (args) => {
  const [show, shouldShow] = useState(args.show);

  useEffect(() => {
    shouldShow(args.show);
  }, [args.show, shouldShow]);

  return (
    <>
      <Loading
        {...{
          show,
          shouldShow,
        }}
      />
      <button
        onClick={() => {
          shouldShow(true);
        }}
      >
        Show
      </button>
    </>
  );
};

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  show: true,
};
