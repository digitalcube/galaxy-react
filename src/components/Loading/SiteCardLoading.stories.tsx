import React from 'react';
import { Meta, Story } from '@storybook/react';
import { SiteCardLoading, SiteCardLoadingProps } from './SiteCardLoading';
const meta: Meta = {
  title: 'Components/Loading/SiteCardLoading',
  component: SiteCardLoading,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<SiteCardLoadingProps> = (args) => {
  return (
    <div className="sites-list">
      <section className="d-flex flex-column site-list-body">
        <SiteCardLoading {...args} />
      </section>
    </div>
  );
};

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  show: true,
};
