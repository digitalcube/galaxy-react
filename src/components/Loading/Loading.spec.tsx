import React from 'react';
import { render } from '@testing-library/react';
import { Default as Loading } from './Loading.stories';

describe('Loading', () => {
  it('to match snapshot (hidden)', () => {
    const { asFragment } = render(
      <Loading
        {...{
          show: false,
          shouldShow: jest.fn(),
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot (open)', () => {
    const { asFragment } = render(
      <Loading
        {...{
          show: true,
          shouldShow: jest.fn(),
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
