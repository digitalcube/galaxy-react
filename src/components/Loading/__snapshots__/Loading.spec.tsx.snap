// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading to match snapshot (hidden) 1`] = `<DocumentFragment />`;

exports[`Loading to match snapshot (open) 1`] = `
<DocumentFragment>
  <div
    class="mr-2 spinner-border"
    role="status"
  >
    <span
      class="sr-only"
    >
      Loading...
    </span>
  </div>
  Loading...
</DocumentFragment>
`;
